alchemist/alchemist-incarnation-protelis/src/main/java/it/unibo/alchemist/model/ProtelisIncarnation.java

```diff
@@ -485,6 +486,10 @@ public void setConcentration(final Molecule mol, final Object c) {
             return notImplemented();
         }
         @Override
+        public GeometricShape<?, ?> getShape() {
+            return null;
```

Comment:
This is going to hurt us violently :)
Rather, in AbstractNode return a punctiform. If this does not extend from the abstract one, make it return a punctiform. `null` is never a good idea.

Reply:
To be honest I saw that getReactions() returned null for this node so I just copied it, I assumed that NoNode was used for some kind of hack anyway.
What about return notImplemented(); like the other methods?
Otherwise I would need to access the environment to get the shapeFactory and build the adimensional shape

Reply:
Sorry, I overlooked that it's NoNode and not ProtelisNode.